docs/internals.md

```diff
@@ -0,0 +1,216 @@
+adjacency list. Each edge and property is stored as an individual cell
+in the row which allows for efficient insertions and deletions. The
+maximum number of cells allowed per row in a particular storage backend
+is therefore also the maximum degree of a vertex that JanusGraph can
+support against this backend.
+
+If the storage backend supports key-order, the adjacency lists will be
+ordered by vertex id, and JanusGraph can assign vertex ids such that the
+graph is effectively partitioned. Ids are assigned such that vertices
+which are frequently co-accessed have ids with small absolute
+difference.
+
+### Individual Edge Layout
+
+![](images/relationlayout.png)
+
+Each edge and property is stored as one cell in the rows of its adjacent
+vertices. They are serialized such that the byte order of the column
+respects the sort key of the edge label. Variable id encoding schemes
+and compressed object serialization are used to keep the storage
+footprint of each edge/cell as small as possible.
+
+Consider the storage layout of an individual edge as visualized in the
+top row of the graphic above. The dark blue boxes represent numbers that
+are encoded with a variable length encoding scheme to reduce the number
+of bytes they consume. Red boxes represent one or multiple property
+values (i.e. objects) that are serialized with compressed meta data
+referenced in the associated property key. Grey boxes represent
+uncompressed property values (i.e. serialized objects).
+
+The serialized representation of an edge starts with the edge label’s
+unique id (as assigned by JanusGraph). This is typically a small number
+and compressed well with variable id encoding. The last bit of this id
+is offset to store whether this is an incoming or outgoing edge. Next,
+the property value comprising the sort key are stored. The sort key is
+defined with the edge label and hence the sort key objects meta data can
+be referenced to the edge label. After that, the id of the adjacent
+vertex is stored. JanusGraph does not store the actual vertex id but the
+difference to the id of the vertex that owns this adjacency list. It is
+likely that the difference is a smaller number than the absolute id and
+hence compresses better. The vertex id is followed by the id of this
+edge. Each edge is assigned a unique id by JanusGraph. This concludes
+the column value of the edge’s cell. The value of the edge’s cell
+contains the compressed serialization of the signature properties of the
+edge (as defined by the label’s signature key) and any other properties
+that have been added to the edge in uncompressed serialization.
+
+The serialized representation of a property is simpler and only contains
+the property’s key id in the column. The property id and the property
+value are stored in the value. If the property key is defined as
+`list()`, however, the property id is stored in the column as well.
+
+## Building JanusGraph
```

Comment:
Here I'm wondering again whether it makes sense to merge these chapters into one. I can't think of a case where someone reads about the JanusGraph data model and then wants to read on on how to build JanusGraph. These topics don't really have anything to do with each other so I wouldn't put them in the same chapter.
(Length of a chapter shouldn't really be a criterion.)